### PR TITLE
sequoia-tool: init at 0.9.0

### PR DESCRIPTION
**Pull request by minijackson**
###### Motivation for this change

A user-friendly PGP tool: https://sequoia-pgp.org/

- The command-line program is actually called "sq" but the Rust package is called "sequoia-tool". I wasn't sure which to choose so I went with the one with the least chance of collision.
- ~~I didn't manage to build it entirely on my laptop because it kept running out of memory / space, so if someone would like to test it locally, or trigger ofborg, I would really appreciate it ^^~~

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/security/sequoia-tool/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  buildInputs = [ capnproto nettle openssl sqlite ];
+
+  LIBCLANG_PATH = libclang + "/lib";
+
+  cargoBuildFlags = [ "--package=sequoia-tool" ];
+
+  cargoSha256 = "03fn0vmaa8axc4yf761ds46p1hxkimbr4dzlbp018rmhgilb96p7";
+
+  meta = with stdenv.lib; {
+    description = "";
```
Comment (risicle) [inline pkgs/tools/security/sequoia-tool/default.nix]: "A user-friendly PGP tool" maybe?

Comment (minijackson) [reply pkgs/tools/security/sequoia-tool/default.nix]: Yes! Must have slipped my mind, sorry about that ^^'

Comment (minijackson) [reply pkgs/tools/security/sequoia-tool/default.nix]: Used a mix of the `Cargo.toml` description and the CLI "about" description

File: pkgs/tools/security/sequoia-tool/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitLab {
+    owner = "sequoia-pgp";
+    repo = "sequoia";
+    rev = "v${version}";
+    sha256 = "19k3vsaz0iqm0qxqadlfsnrnprn4lrb8078mm2if3gq7klb5csjb";
+  };
+
+  nativeBuildInputs = [ pkgconfig clang libclang ];
+  buildInputs = [ capnproto nettle openssl sqlite ];
```
Comment (risicle) [inline pkgs/tools/security/sequoia-tool/default.nix]: To get this building under darwin, add `++ stdenv.lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Security` to the `buildInputs`.

Comment (minijackson) [reply pkgs/tools/security/sequoia-tool/default.nix]: Thank you!

**Comment by risicle** [comment]
Builds for me on non-nixos linux x86_64 too. I've pushed the resultant closure of `/nix/store/yfgbng5kkp68kghq48l2925cklhqkidd-sequoia-tool-0.8.0` up to https://ris.cachix.org/ if you want to inspect it.

**Comment by minijackson** [comment]
Thanks a lot @risicle! I've played with it a bit, and it seems to run quite fine ^^

**Comment by sauyon** [comment]
Works for me on Arch, what needs to be done to get this merged?

**Comment by minijackson** [comment]
Some attention, I think ^^

**Comment by worldofpeace** [comment]
Thank you @minijackson.

If you ever feel like a change looks good, has all the approvals, and you feel meets the marks, feel free to just ping me.

**Comment by minijackson** [comment]
Thank you @worldofpeace !

**Pull request merged**